### PR TITLE
add wait group to rebalance and get collateral health check

### PR DESCRIPTION
**Pull request by athiwatp**
add wait group to rebalance and get collateral health check for improve time to call horizon testnet, change horizon testnet from stellar to velo horizon testnet server improve time to call get assets

### CODE REVIEW

File: node/app/layers/usecases/get_collateral_health_check.go [hunk 1]
```diff
@@ -56,62 +57,80 @@ func (useCase *useCase) GetCollateralHealthCheck(ctx context.Context) (*entities
...
+					collateralPerCredit = stableAmount.Mul(issuerAccount.PeggedValue).Div(medianPriceSgd)
+				case vxdr.CurrencyUSD:
+					collateralPerCredit = stableAmount.Mul(issuerAccount.PeggedValue).Div(medianPriceUsd)
+				default:
+					panic(nerrors.ErrInternal{Message: constants.ErrPeggedCurrencyIsNotSupport})
+				}
+				// sum total drs collateral required amount of tp
+				collateralPerTp = collateralPerTp.Add(collateralPerCredit)
 			}
+			// sum total drs collateral required amount amount
```
Comment (jura-b) [inline node/app/layers/usecases/get_collateral_health_check.go]: amount

File: node/app/layers/usecases/get_collateral_health_check.go [hunk 2]
```diff
@@ -56,62 +57,80 @@ func (useCase *useCase) GetCollateralHealthCheck(ctx context.Context) (*entities
...
 			if err != nil {
-				return nil, nerrors.ErrPrecondition{Message: errors.Wrapf(err, constants.ErrGetAsset, assetDetail[0]).Error()}
+				panic(nerrors.ErrPrecondition{
+					Message: errors.Wrap(err, constants.ErrGetTrustedPartnerMetaAccountDetail).Error(),
+				})
 			}
 
-			if len(assetPage.Embedded.Records) < 1 {
-				continue
+			// calculate drs collateral required amount amount per tp
```
Comment (jura-b) [inline node/app/layers/usecases/get_collateral_health_check.go]: amount

File: node/app/layers/usecases/get_collateral_health_check.go [hunk 3]
```diff
@@ -56,62 +57,80 @@ func (useCase *useCase) GetCollateralHealthCheck(ctx context.Context) (*entities
 
 	var drsCollateralRequiredAmount = decimal.Zero
 
+	var wg sync.WaitGroup
+	wg.Add(len(tpListData))
+	var wgErr nerrors.NodeError
+
 	// calculate drs collateral required amount amount
```
Comment (jura-b) [inline node/app/layers/usecases/get_collateral_health_check.go]: amount

File: node/app/layers/usecases/get_collateral_health_check.go [hunk 4]
```diff
@@ -56,62 +57,80 @@ func (useCase *useCase) GetCollateralHealthCheck(ctx context.Context) (*entities
 
 	var drsCollateralRequiredAmount = decimal.Zero
 
+	var wg sync.WaitGroup
```
Comment (Kafakk) [inline node/app/layers/usecases/get_collateral_health_check.go]:
```suggestion
	var waitGroup sync.WaitGroup
```

**Review by jura-b — APPROVED** [review]

**Pull request merged**